app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt

```diff
@@ -708,6 +708,44 @@ interface ProjectViewModel {
                     .subscribe(this.heartDrawableId)
 
             //Tracking
+            val refTagsAndProject = Observable.combineLatest<RefTag, RefTag, Project, RefTagsAndProject>(refTag, cookieRefTag, currentProject)
+            { refTagFromIntent, refTagFromCookie, project -> RefTagsAndProject(refTagFromIntent, refTagFromCookie, project) }
+                    .filter { it.project.hasRewards() }
+                    .take(1)
```

Comment:
Do we need the `take(1)` with the `takeWhen` further down?

Reply:
I think so because `currentProject` emits again when the `Project` gets refreshed and we don't want to track this screen view again for those refreshes.

Reply:
Oh right - I guess in this case it wouldn't matter because the event doesn't track based on the screen refreshing but on the button press. Either way 🤷‍♂